packages/flutter/lib/src/cupertino/page_scaffold.dart

```diff
@@ -75,12 +75,18 @@ class CupertinoPageScaffold extends StatefulWidget {
 }
 
 class _CupertinoPageScaffoldState extends State<CupertinoPageScaffold> {
-  final ScrollController _primaryScrollController = ScrollController();
+  ScrollController? _primaryScrollController;
+
+  @override
+  void didChangeDependencies() {
+    _primaryScrollController  = PrimaryScrollController.of(context);
```

Comment:
Could you just look up the controller in _handleStatusBarTap just-in-time instead of caching it?

Reply:
Sure!